doc/release/upcoming_changes/25149.c_api.rst

```diff
@@ -0,0 +1,10 @@
+Lager ``NPY_MAXDIMS`` and ``NPY_MAXARGS``, ``NPY_RAVEL_AXIS`` introduced
+------------------------------------------------------------------------
+
+``NPY_MAXDIMS`` is now 64, you may want to review its use.  This is usually
+used in stack allocation, where the increase should be safe.
+However, we do encourage generally to remove any use of ``NPY_MAXDIMS`` and
+``NPY_MAXARGS`` to eventually allow removing the constraint completely.
+In some cases, ``NPY_MAXDIMS`` was passed (and returned) to mean ``axis=None``
```

Comment:
Can you make "In some cases" a little more specific? Does this only impact things that interact with numpy via the C API, or could you trigger this behavior from the python API too?